components/brave_rewards/browser/database/database_recurring_tip.cc

```diff
@@ -0,0 +1,181 @@
+        "amount DOUBLE DEFAULT 0 NOT NULL,"
+        "added_date INTEGER DEFAULT 0 NOT NULL,"
+        "CONSTRAINT fk_%s_publisher_id"
+        "    FOREIGN KEY (publisher_id)"
+        "    REFERENCES publisher_info (publisher_id)"
+        "    ON DELETE CASCADE"
+      ")",
+      table_name_,
+      table_name_);
+
+  return db->Execute(query.c_str());
+}
+
+bool DatabaseRecurringTip::CreateIndex(sql::Database* db) {
+  return CreateIndexV2(db);
+}
+
+bool DatabaseRecurringTip::CreateIndexV2(sql::Database* db) {
+  return this->InsertIndex(db, table_name_, "publisher_id");
+}
+
+bool DatabaseRecurringTip::Migrate(sql::Database* db, const int target) {
+  switch (target) {
+    case 2: {
+      return MigrateToV2(db);
+    }
+    default: {
+      NOTREACHED();
+      return false;
+    }
+  }
+}
+
+bool DatabaseRecurringTip::MigrateToV2(sql::Database* db) {
+  if (!CreateTableV2(db)) {
+    return false;
+  }
+
+  if (!CreateIndexV2(db)) {
+    return false;
+  }
+
+  return true;
+}
+
+bool DatabaseRecurringTip::InsertOrUpdate(
+    sql::Database* db,
+    ledger::RecurringTipPtr info) {
+  if (!info || info->publisher_key.empty()) {
+    return false;
+  }
+
+  const std::string query = base::StringPrintf(
+      "INSERT OR REPLACE INTO %s "
+      "(publisher_id, amount, added_date) "
+      "VALUES (?, ?, ?)",
+      table_name_);
+
+  sql::Statement statement(
+      db->GetCachedStatement(SQL_FROM_HERE, query.c_str()));
+
+  statement.BindString(0, info->publisher_key);
+  statement.BindDouble(1, info->amount);
+  statement.BindInt64(2, info->created_at);
+
+  return statement.Run();
+}
+void DatabaseRecurringTip::GetAllRecords(
+    sql::Database* db,
+    ledger::PublisherInfoList* list) {
+  if (!list) {
+    return;
+  }
+
+  const std::string query = base::StringPrintf(
+    "SELECT pi.publisher_id, pi.name, pi.url, pi.favIcon, "
+    "rd.amount, rd.added_date, spi.status, pi.provider "
+    "FROM %s as rd "
+    "INNER JOIN publisher_info AS pi ON rd.publisher_id = pi.publisher_id "
+    "LEFT JOIN server_publisher_info AS spi "
+    "ON spi.publisher_key = pi.publisher_id ",
+    table_name_);
+
+  sql::Statement statement(db->GetUniqueStatement(query.c_str()));
+
+  while (statement.Step()) {
+    auto publisher = ledger::PublisherInfo::New();
+    publisher->id = statement.ColumnString(0);
+    publisher->name = statement.ColumnString(1);
+    publisher->url = statement.ColumnString(2);
+    publisher->favicon_url = statement.ColumnString(3);
+    publisher->weight = statement.ColumnDouble(4);
+    publisher->reconcile_stamp = statement.ColumnInt64(5);
+    publisher->status =
+        static_cast<ledger::mojom::PublisherStatus>(statement.ColumnInt64(6));
+    publisher->provider = statement.ColumnString(7);
+
+    list->push_back(std::move(publisher));
+  }
+}
+
+bool DatabaseRecurringTip::DeleteRecord(
+    sql::Database* db,
+    const std::string& publisher_key) {
+  const std::string query = base::StringPrintf(
```

Comment:
`DCHECK(db)` and `if (!db) {...` to identify potential issues when running debug builds

Reply:
covered in https://github.com/brave/brave-browser/issues/7145